actionview/lib/action_view/helpers/form_helper.rb

```diff
@@ -720,6 +961,235 @@ def fields_for(record_name, record_object = nil, options = {}, &block)
+      #
+      #   # Using +fields+ with +form_with+:
+      #   form_with model: @post do |form|
+      #     form.text_field :title
+      #
+      #     form.fields :comment do |fields|
+      #       fields.text_field :body
+      #     end
+      #   end
+      #
+      # Much like +form_with+ a FormBuilder instance associated with the scope
+      # or model is yielded, so any generated field names are prefixed with
+      # either the passed scope or the scope inferred from the <tt>:model</tt>.
+      #
+      # === Mixing with other form helpers
+      #
+      # While +form_with+ uses a FormBuilder object it's possible to mix and
+      # match the stand-alone FormHelper methods and methods
+      # from FormTagHelper:
+      #
+      #   fields model: @comment do |fields|
+      #     fields.text_field :body
+      #
+      #     text_area :commenter, :biography
+      #     check_box_tag "comment[all_caps]", "1", @comment.commenter.hulk_mode?
+      #   end
+      #
+      # Same goes for the methods in FormOptionHelper and DateHelper designed
+      # to work with an object as a base, like
+      # FormOptionHelper#collection_select and DateHelper#datetime_select.
+      #
+      # === Nested Attributes Examples
+      #
+      # When the current scope's object has a nested attribute writer —
+      # most likely through +accepts_nested_attributes_for+ — for an
+      # attribute, +fields+ yields a new scope for that attribute. Useful when
+      # setting attributes on a parent and its associations in one go.
+      #
+      # Writers are considered a nested attributes setter if they're of the
+      # <tt>*_attributes=</tt> form, e.g. <tt>address_attributes=</tt>.
+      #
+      # Depending on the association's reader method return value a different
+      # form builder is yielded. For single object returns a one-to-one builder,
+      # while a one-to-many builder for array returns.
+      #
+      # ==== One-to-one
+      #
+      # Consider a Person class which returns a _single_ Address from the
+      # <tt>address</tt> reader method and responds to the
+      # <tt>address_attributes=</tt> writer method:
+      #
+      #   class Person
+      #     def address
+      #       @address
+      #     end
+      #
+      #     def address_attributes=(attributes)
+      #       # Process the attributes hash
+      #     end
+      #   end
+      #
+      # This model can now be used with a nested +fields+, like so:
+      #
+      #   form_with model: @person do |person_form|
+      #     ...
+      #     <%= person_form.fields_for :address do |address_fields| %>
+      #       Street  : <%= address_fields.text_field :street %>
+      #       Zip code: <%= address_fields.text_field :zip_code %>
+      #     <% end %>
+      #     ...
+      #   <% end %>
+      #
+      # When address is already an association on a Person you can use
+      # +accepts_nested_attributes_for+ to define the writer method for you:
```

Comment:
Oh yeah! 🔥 Reading through this I thought `accepts_nested_attributes_for` stuck out, refrained from doing anything because I didn't have a better idea yet. But yes, let's start down the path toward something better!